Questions.md

```diff
@@ -0,0 +1,9 @@
+Something what i don't understand?
+
+- For example we want to push something with a sql plugin to a database. Why we first write it down to disk and then read in memory with pandas and then push it further?    
+
+
+
+Future:
+
+- If we make an external materialization we want that upstream models can refer to it? How to do it we have to register an df and create an view to it?
```

Comment:
The current `external` materialization does this already; dbt-duckdb creates a `VIEW` in the DuckDB database that points at the externally materialized file so that it can be used/queried like normal by any other models that we run.

Reply:
This is a bit tricky because the underlying delta table can't be just created but it has to be registered first as df. We do that currently when the delta table is defined in the source but we have to invoke this process here too. Let me try some stuff here if not we can make it a limitation and provide as the last layer that can't be referenced

Reply:
As of my understanding, if you use SQL/excel write plugin, you cant reference it afterward?

Reply:
Yes, you can reference any `external` materialization after it gets created in other, downstream models (sorry for the lag I've been traveling in Asia and will be here for the next few weeks)

Reply:
Just noting that I would support a custom materialization type for `delta` tables if that is what we really need to make the write-side work well here.

Reply:
i think that we can also put it in external but i have to refactor it a bit, but i am still not so far away, currently i just play around and try to make it work anyway